shared-module/usb_cdc/__init__.c

```diff
@@ -259,3 +264,154 @@ bool common_hal_usb_cdc_enable(bool console, bool data) {
 
     return true;
 }
+
+#if CIRCUITPY_USB_VENDOR
+#include "usb_vendor_descriptors.h"
+
+#define BOS_TOTAL_LEN      (TUD_BOS_DESC_LEN + TUD_BOS_WEBUSB_DESC_LEN + TUD_BOS_MICROSOFT_OS_DESC_LEN)
+
+#define MS_OS_20_DESC_LEN  0xB2
+
+// BOS Descriptor is required for webUSB
+uint8_t const desc_bos[] =
```

Comment:
@dhalbert @hathach do either of you have something to say about whether this is right or better done in another way? I'm not familiar with webusb so I don't understand what is going on here.

Reply:
The USB Vendor stuff was added by a contributor for 6.x. I did not attempt to redo this for 7, and I assumed it needed to be reworked.

WebUSB is more complicated than WebSerial, and I'd recommend trying to use that if possible.

@bsatrom Did you test webUSB? Are you using it?

Reply:
Hi there @dhalbert! We have not yet done thorough testing, but I have seen the device accessible from the browser - I plan to circle back on this over the coming days. 

Because it is conditional code, I thought it would be ok including it so others can build on what is done here, but I'm happy to remove those changes if they are blocking approval of the PR. They are not needed for this board at present.

Reply:
Thanks - I'd say leave it in, maybe with a comment caveat that it's not fully tested, since we only turn this on for one other board right now.